libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -386,15 +388,19 @@ - (void) syncUp:(SFSyncState*)sync updateSync:(SyncUpdateBlock)updateSync failSy
     }
     
     // Fail block for rest call
-    SFRestFailBlock failRest = ^(NSError *error) {
-        failSync(@"REST call failed", error);
+    void (^failBlock)(NSError *error) = ^(NSError *error) {
+        failSync(@"Server call failed", error);
     };
 
     // Otherwise, there's work to do.
-    [self syncUpOneEntry:sync recordIds:dirtyRecordIds index:0 updateSync:updateSync failRest:failRest];
+    [self syncUpOneEntry:sync recordIds:dirtyRecordIds index:0 updateSync:updateSync failBlock:failBlock];
 }
 
-- (void) syncUpOneEntry:(SFSyncState*)sync recordIds:(NSArray*)recordIds index:(NSUInteger)i updateSync:(SyncUpdateBlock)updateSync failRest:(SFRestFailBlock)failRest {
+- (void)syncUpOneEntry:(SFSyncState*)sync
+             recordIds:(NSArray*)recordIds
+                 index:(NSUInteger)i
+            updateSync:(SyncUpdateBlock)updateSync
+             failBlock:(void (^)(NSError *))failBlock {
```

Comment:
We could do that.  I had initially focused on decoupling the sync manager from the SFRest functionality, but we can add local typedefs to keep parity with `SyncUpdateBlock`.